### PR TITLE
Add SQLite and UI Table

### PR DESCRIPTION
**Pull request by j-dunham**
## What is changing
- Now saves prompts to a local SQLite file
- Loads saved prompts into a Table and displays it when `CTRL+T` is pressed.  A prompt can be added to the prompt view port using the arrow keys and pressing `ENTER`. 

I am imaging that in a future change the current prompts in the view port can be sent as context to OpenAI and being able to add previous prompts will be useful then

### CODE REVIEW

File: services/storage/db.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	tableSql := fmt.Sprintf("SELECT name FROM sqlite_master WHERE type='table' AND name='%s'", table)
+	rows := query_db(tableSql)
+	defer rows.Close()
+
+	exists := rows.Next()
+	if exists {
+		return
+	}
+
+	createSql := fmt.Sprintf("CREATE TABLE %s (id INTEGER PRIMARY KEY AUTOINCREMENT, prompt TEXT, response TEXT, CreatedAt DATETIME DEFAULT CURRENT_TIMESTAMP)", table)
```
Comment (simonsanchez) [inline services/storage/db.go]: Alternatively, looks like sqlite supports `IF NOT EXISTS`

```suggestion
        createSql := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (id INTEGER PRIMARY KEY AUTOINCREMENT, prompt TEXT, response TEXT, CreatedAt DATETIME DEFAULT CURRENT_TIMESTAMP)", table)
```
[Found this in a blog](https://www.sqlitetutorial.net/sqlite-create-table/).

File: services/storage/db.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+func execute_sql(sql string, args ...interface{}) (rows sql.Result, err error) {
+	db := openDB()
+	defer db.Close()
+
+	res, err := db.Exec(sql, args...)
+	if err != nil {
+		err = fmt.Errorf("execute_sql: %s .. %s", err, sql)
+		log.Fatal(err)
```
Comment (simonsanchez) [inline services/storage/db.go]: Should the CLI hard fail if there is an error executing a SQL command?

File: main.go [hunk 3]
```diff
@@ -56,27 +92,54 @@ func initialModel() model {
...
 		responseStyle: lipgloss.NewStyle().Foreground(lipgloss.Color("3")),
 		err:           nil,
+		table:         setupTable(),
+		showTable:     false,
+		help:          setupHelp(),
 	}
 }
 
+func savePrompt(prompt string, response string) {
+	storage.CreateTable()
```
Comment (simonsanchez) [inline main.go]: This line can be removed since L23 already calls `storage.CreateTable`. 
```suggestion
```

**Review by simonsanchez — APPROVED** [review]

File: services/storage/db.go (hunk 2 above)
Comment (j-dunham) [reply services/storage/db.go]: Yeah it makes sense to not fail on DB operations

Comment (j-dunham) [reply services/storage/db.go]: updated!

**Pull request merged**